crates/db_schema/src/source/oauth_provider.rs

```diff
@@ -0,0 +1,153 @@
+  /// The client_secret is provided by the OAuth 2.0 provider and is used to authenticate this
+  /// service with the provider
+  #[serde(skip)]
+  pub client_secret: SensitiveString,
+  /// Lists the scopes requested from users. Users will have to grant access to the requested scope
+  /// at sign up.
+  pub scopes: String,
+  /// Automatically sets email as verified on registration
+  pub auto_verify_email: bool,
+  /// Allows linking an OAUTH account to an existing user account by matching emails
+  pub account_linking_enabled: bool,
+  /// switch to enable or disable an oauth provider
+  pub enabled: bool,
+  pub published: DateTime<Utc>,
+  pub updated: Option<DateTime<Utc>>,
+}
+
+#[skip_serializing_none]
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "full", derive(TS))]
+#[cfg_attr(feature = "full", ts(export))]
+pub struct OAuthProvider {
+  pub id: OAuthProviderId,
+  /// The OAuth 2.0 provider name displayed to the user on the Login page
+  pub display_name: String,
+  /// The issuer url of the OAUTH provider.
+  #[cfg_attr(feature = "full", ts(type = "string"))]
+  pub issuer: Option<DbUrl>,
+  /// The authorization endpoint is used to interact with the resource owner and obtain an
+  /// authorization grant. This is usually provided by the OAUTH provider.
+  #[cfg_attr(feature = "full", ts(type = "string"))]
+  pub authorization_endpoint: DbUrl,
+  /// The token endpoint is used by the client to obtain an access token by presenting its
+  /// authorization grant or refresh token. This is usually provided by the OAUTH provider.
+  #[cfg_attr(feature = "full", ts(type = "string"))]
+  pub token_endpoint: Option<DbUrl>,
+  /// The UserInfo Endpoint is an OAuth 2.0 Protected Resource that returns Claims about the
+  /// authenticated End-User. This is defined in the OIDC specification.
+  #[cfg_attr(feature = "full", ts(type = "string"))]
+  pub userinfo_endpoint: Option<DbUrl>,
+  /// The OAuth 2.0 claim containing the unique user ID returned by the provider. Usually this
+  /// should be set to "sub".
+  pub id_claim: Option<String>,
+  /// The OAuth 2.0 claim containing the user name returned by the provider. This depends on the
+  /// provider and could be "name", "username", ...
+  pub name_claim: Option<String>,
+  /// The client_id is provided by the OAuth 2.0 provider and is a unique identifier to this
+  /// service
+  pub client_id: String,
+  /// Lists the scopes requested from users. Users will have to grant access to the requested scope
+  /// at sign up.
+  pub scopes: String,
+  /// Automatically sets email as verified on registration
+  pub auto_verify_email: Option<bool>,
+  /// Allows linking an OAUTH account to an existing user account by matching emails
+  pub account_linking_enabled: Option<bool>,
+  /// switch to enable or disable an oauth provider
+  pub enabled: Option<bool>,
+  pub published: Option<DateTime<Utc>>,
+  pub updated: Option<DateTime<Utc>>,
+}
```

Comment:
This duplicate struct is unnecessary, becuase `serde(skip)` already ensures that the secret is not sent to clients. Its the same attribute we use for `local_user.password`.

Reply:
Looks fine now, please add a comment to describe what `PublicOAuthProvider` is for. Basically what you wrote in the other discussion above.